Universidad Torcuato Di Tella

Licenciatura en Tecnología Digital\
**Tecnología Digital VI: Inteligencia Artificial**

# **Taller de análisis de datos**

Resolver las siguientes consignas.

### **Consigna 01**

> En un *data frame* llamado `vuelos`, cargar los datos que se encuentran en el archivo `td6-p02-c-datos-vuelos.txt`, disponible en el Campus Virtual, y observarlos.

In [ ]:
import pandas as pd

# Cargar el archivo en un data frame
# Parámetros: archivo y el separador
vuelos = pd.read_csv('td6-p02-c-datos-vuelos.txt', sep=';')

### **Consigna 02**

> ¿Datos de cuántos vuelos tiene el data frame `vuelos`?

In [3]:
num_vuelos = vuelos.shape[0]
print(num_vuelos)

3318080


### **Consigna 03**

> ¿Cuántas columnas tiene el data frame `vuelos`?

In [18]:
num_columnas = vuelos.shape[1]
print(num_columnas)

# pregunta: cuál es la diferencia del índice?
# son las columnas, no las filas

28


### **Consigna 04**

> Considerar un vuelo como demorado si el valor de `ArrDelay` es superior a 20 min. Teniendo esto en cuenta,
>
> * ¿qué aerolínea (*tip*: aprovechar el diccionario de variables e Internet) tiene la mayor proporción de vuelos demorados?
>
> * Y, ¿en qué aeropuerto de destino es más común (en términos relativos) que haya demoras en el arribo?
>
> *Tip*: qué significa cada código lo pueden encontrar en la siguiente página: https://www.iata.org/en/publications/directories/code-search/.

In [16]:
vuelos_demorados = vuelos[vuelos["ArrDelay"] > 20]
vuelos_demorados["UniqueCarrier"].value_counts()/vuelos["UniqueCarrier"].value_counts()

UniqueCarrier
AA    0.220650
MQ    0.191959
OO    0.149205
US    0.141364
WN    0.144385
Name: count, dtype: float64

**Respuesta**: AA.

In [17]:
dest_demorados = vuelos_demorados["Dest"].value_counts()/vuelos["Dest"].value_counts()
dest_demorados.sort_values(ascending=False)

Dest
OTH    0.365571
EGE    0.356979
GUC    0.333333
MQT    0.326284
EWR    0.310471
         ...   
WYS    0.060606
BLI    0.057851
IYK    0.040881
CYS         NaN
OGD         NaN
Name: count, Length: 230, dtype: float64

**Respuesta**: OTH.

### **Consigna 05**

> ¿Es más o menos común que haya vuelos demorados los fines de semana respecto a los días de semana?

In [ ]:
demorados = vuelos[vuelos["ArrDelay"] > 20]
demorados["weekend"] = (vuelos["DayOfWeek"] == 6) | (vuelos["DayOfWeek"] == 7)

cant_vuelos = vuelos.copy()
cant_vuelos["weekend"] = (vuelos["DayOfWeek"] == 6) | (vuelos["DayOfWeek"] == 7)

semana_proporciones = pd.merge(demorados["weekend"].value_counts(), cant_vuelos["weekend"].value_counts())


**Respuesta**: COMPLETAR.

### **Consigna 06**

> Para los casos en que hay minutos de demora debido al clima, ¿cuál es la duración promedio de esas demoras? *Tip*: piensen si, para resolver esta consigna, tiene o no sentido considerar los ceros.

In [19]:
vuelos[vuelos["WeatherDelay"] > 0][["WeatherDelay"]].mean()

WeatherDelay    41.220588
dtype: float64

**Respuesta**: 41.220588

### **Consigna 07**

> * ¿Cuáles son el número de vuelo, el origen y el destino correspondientes al vuelo más largo, en términos de tiempo, que se registra en los datos?
>
> * ¿Cuál es la duración, en horas, de dicho vuelo?

In [20]:
vuelos.columns

Index(['Year', 'Month', 'DayofMonth', 'DayOfWeek', 'DepTime', 'CRSDepTime',
       'ArrTime', 'CRSArrTime', 'UniqueCarrier', 'FlightNum', 'TailNum',
       'ActualElapsedTime', 'CRSElapsedTime', 'AirTime', 'ArrDelay',
       'DepDelay', 'Origin', 'Dest', 'TaxiIn', 'TaxiOut', 'Cancelled',
       'CancellationCode', 'Diverted', 'CarrierDelay', 'WeatherDelay',
       'NASDelay', 'SecurityDelay', 'LateAircraftDelay'],
      dtype='object')

**Respuesta**: COMPLETAR.

In [ ]:
vuelos[vuelos["ActualElapsedTime"] == max(vuelos["ActualElapsedTime"])]

**Respuesta**: COMPLETAR.

### **Consigna 08**

> ¿Cuál es la cantidad de vuelos que se realiza para cada combinación origen-destino? *Tip*: guardar esta información porque es necesaria para resolver la siguiente consigna.

In [ ]:
cant_vuelos_origen_destino = pd.crosstab(vuelos["Origin"], vuelos["Dest"])
cant_vuelos_origen_destino

**Respuesta**: COMPLETAR.

### **Consigna 09**

> ¿Cuál es el tramo origen-destino para el que se hicieron más vuelos? *Tip*: guardar también esta información porque es necesaria para resolver la siguiente consigna.

In [ ]:
# axis es la columna a la cual se le aplica max()
cant_vuelos_origen_destino.max(axis = 1).max()
print(cant_vuelos_origen_destino.max(axis = 0).idxmax()) # Axis = 0 -> columna (Destino)
print(cant_vuelos_origen_destino.max(axis = 1).idxmax()) # Axis = 1 -> fila (Origen)

**Respuesta**: COMPLETAR.

### **Consigna 10**

> * Considerando los vuelos correspondientes al tramo más frecuente y que no fueron cancelados, realizar un diagrama de cajas que permita ver la distribución de la duración de los vuelos.
>
> * Una vez hecho el diagrama, ¿a qué consideran que pueden deberse los valores extremos que se visualizan en él?

In [ ]:
diagrama_cajas_datos = vuelos.loc[(vuelos["Origin"] == tramo)]

**Respuesta**: COMPLETAR.

### **Consigna 11**

> Hacer una figura que permita ver si la distribución de la duración de los vuelos considerados en la consigna anterior varía o no en función de si ocurre un día o de semana o de fin de semana y comentarla.

In [ ]:
plt.ioff()
diagrama_cajas_datos.boxplot(column = )

**Respuesta**: COMPLETAR.

### **Consigna 12**

> Hacer una figura que permita ver, por mes, cuántos vuelos tienen demora en la salida y cuántos, en el arribo y comentarla.

In [ ]:
# COMPLETAR.

**Respuesta**: COMPLETAR.